src/main/java/proj/kedabra/billsnap/business/service/impl/UserDetailsServiceImpl.java

```diff
@@ -51,6 +51,10 @@ private UserDetails toUserDetails(Account user) {
             authorities.add(new SimpleGrantedAuthority(bill.getId().toString()));
         });
 
+        user.getNotifications().forEach(notification ->
+                authorities.add(new SimpleGrantedAuthority("INVITATION_" + notification.getId()))
```

Comment:
If you really want to add this new role, what I suggest isn't using notifications (because you could have answered notifications). What you should do is: 

`user.getBills().stream().filter(ab -> InvitationStatusEnum.PENDING == ab.getStatus).map(AccountBill::getBill).map(Bill::getId).map(Object::toString).map("INVITATION_"::concat).map(SimpleGrantedAuthority::new).forEach(authorities::add); `

Additionally, you shouldn't be using `notification.id` here when in the controller you're passing `bill.id`

Reply:
good catch, it should only be bill.id

Reply:
I'll remove this role, it's redundant